src/com/hms_networks/americas/sc/historicaldata/HistoricalDataQueueManager.java

```diff
@@ -163,7 +163,7 @@ public static synchronized ArrayList getFifoNextSpanData(
      * end time is in the future.
      */
     long startTimeTrackerMsPlusSpan = startTimeTrackerMsLong + getQueueFifoTimeSpanMillis();
-    long endTimeTrackerMsLong = Math.min(startTimeTrackerMsPlusSpan, System.currentTimeMillis());
+    long endTimeTrackerMsLong = Math.min(startTimeTrackerMsPlusSpan, getCurrentTimeWithOffset());
```

Comment:
So i think i ran into this myself and ended up working around it by setting my time to utc. you may want to contact the customer and warn them about this.

Reply:
I agree. I was going to reach out to Tom and see if he is ok  with maybe giving them a v1.1, since this can affect promised functionality